app/components/avo/actions_component.rb

```diff
@@ -1,34 +1,37 @@
 # frozen_string_literal: true
 
-class Avo::ActionsComponent < ViewComponent::Base
+class Avo::ActionsComponent < Avo::BaseComponent
   include Avo::ApplicationHelper
-  attr_reader :label, :size, :as_row_control
-
-  def initialize(actions: [], resource: nil, view: nil, exclude: [], include: [], style: :outline, color: :primary, label: nil, size: :md, as_row_control: false, icon: nil)
-    @actions = actions || []
-    @resource = resource
-    @view = view
-    @exclude = Array(exclude)
-    @include = include
-    @color = color
-    @style = style
-    @label = label || I18n.t("avo.actions")
-    @size = size
-    @icon = icon
-    @as_row_control = as_row_control
+
+  prop :actions, _Array(Avo::BaseAction), default: -> { [] }, reader: :public
+  prop :resource, _Nilable(Avo::BaseResource)
+  prop :view, _Nilable(String)
+  prop :exclude, Array, default: -> { [] } do |value|
+    Array(value)
+  end
+  prop :include, Array, default: -> { [] }
+  prop :style, Symbol, default: :outline
+  prop :color, Symbol, default: :primary
+  prop :label, _Nilable(String), reader: :public do |value|
+    value || I18n.t("avo.actions")
+  end
+  prop :size, Symbol, default: :md, reader: :public
```

Comment:
I’m not sure what the union of sizes is for this component. I couldn't see that this option is even being used at the moment.